core/capability-negotiation.md

```diff
@@ -0,0 +1,457 @@
+| `302` | `cap-notify` | This capability is enabled implicitly with `302`, and adds the `CAP NEW` and `CAP DEL` messages which let the client know about added and removed capabilities. |
+
+#### Capability Values
+
+If the client supports CAP Version `302`, the server MAY specify additional data for each
+capability using the `<name>=<value>` format in `CAP LS` and `CAP NEW` replies.
+
+Each capability, if it supports a value, defines what this value means in its specification.
+
+Example:
+
+    Client: CAP LS 302
+    Server: CAP * LS :multi-prefix sasl=PLAIN,EXTERNAL server-time draft/packing=EX1,EX2
+
+#### Multiline replies to `CAP LS` and `CAP LIST`
+
+Servers MAY send multiple lines in response to `CAP LS`. If the reply contains multiple
+lines (due to IRC line length limitations), and the client supports CAP Version `302`,
+all but the last reply MUST have a parameter containing only an asterisk (`*`) preceding
+the capability list. This lets clients know that more CAP lines are incoming, so that it
+can delay capability negotiation until it has seen all available server capabilities.
+
+If a multi-line response to `CAP LIST` is sent, and the client supports CAP Version `302`,
+the server MUST delimit all replies except for the last one sent as noted above.
+
+Example with a client that **does not** support CAP version `302`:
+
+    Client: CAP LS
+    Server: CAP * LS :multi-prefix extended-join account-notify batch invite-notify tls
+    Server: CAP * LS :cap-notify server-time example.org/dummy-cap=dummyvalue example.org/second-dummy-cap
+    Server: CAP * LS :userhost-in-names sasl=EXTERNAL,DH-AES,DH-BLOWFISH,ECDSA-NIST256P-CHALLENGE,PLAIN
+
+Example with a client that supports CAP version `302`:
+
+    Client: CAP LS 302
+    Server: CAP * LS * :multi-prefix extended-join account-notify batch invite-notify tls
+    Server: CAP * LS * :cap-notify server-time example.org/dummy-cap=dummyvalue example.org/second-dummy-cap
+    Server: CAP * LS :userhost-in-names sasl=EXTERNAL,DH-AES,DH-BLOWFISH,ECDSA-NIST256P-CHALLENGE,PLAIN
+
+Example of a `LIST` reply with a client that supports CAP version `302`:
+
+    Client: CAP LIST
+    Server: CAP modernclient LIST * :example.org/example-cap example.org/second-example-cap account-notify
+    Server: CAP modernclient LIST :invite-notify batch example.org/third-example-cap
+
+### The CAP LIST subcommand
+
+The LIST subcommand is used to list the capabilities associated with the active connection.
+The client should send a LIST subcommand with no other arguments to solicit a list of
+active capabilities.
+
+If no capabilities are active, an empty parameter must be sent.
+
+Example:
+
+    Client: CAP LIST
+    Server: CAP * LIST :multi-prefix
+
+### The CAP REQ subcommand
+
+The REQ subcommand is used to request a change in capabilities associated with the active
+connection.  Its sole parameter must be a list of space-separated capability identifiers.
```

Comment:
The wording for the description of the sole parameter here should be unified between LS and LIST, REQ, ACK, NAK, NEW, DEL.

LS and LIST don't mention the server sent aspect at all, except by chance in the example. NEW and DEL describe the format as "last parameter" and provide an explicit example, REQ and ACK describe the "sole parameter", NAK describes the "argument". I'd favour NEW/DEL style.

Reply:
Unified the language here, should be more consistent now.